soroban-env-host/src/budget.rs

```diff
@@ -584,6 +584,67 @@ impl Display for BudgetImpl {
     }
 }
 
+#[allow(unused)]
+#[cfg(test)]
+impl BudgetImpl {
+    fn retrieve_unscaled_param(param: ScaledU64, is_protocol20: bool) -> u64 {
+        const PROTOCOL_20_LIN_TERM_SCALE_BITS: u32 = 7;
+        match is_protocol20 {
+            true => param.unscale() << PROTOCOL_20_LIN_TERM_SCALE_BITS,
```

Comment:
Why do we need this check?

Reply:
It's not needed now since we didn't change the number of bits. But just wanted to be extra sure we are scaling to the correct bits, in case we did decide to change it (now or in the future).

Reply:
But is there an practical need to do that? Even then, we do have extensions, so it would be more reasonable to put the scaling bits there (though I'm not sure why would we ever want to do that; 128 seems good enough even for fast operations).

Reply:
Removed it. Realistically we will rarely need to print a current set of parameters in an older protocol. So I removed it for less confussion. The whole things is gated by `test` anyway .

Reply:
Yeah, the test gating was actually confusing; ideally tests wouldn't need to deal with protocol versions much.

FWIW the parameter set should normally be independent of the protocol. We might introduce the new cost parameters, but otherwise it doesn't really matter from the host test perspective what the exact costs are. So I would try to limit the impact of the calibration changes on the host (see comment below). Maybe it's worth to have just a sanity check test that exercises all the newly calibrated costs and dumps the results (we do have one such test in hostile.rs, according to diffs), and otherwise have the calibration constant.